fda/registration.py

```diff
@@ -0,0 +1,490 @@
+"""Registration of functional data module.
+
+This module contains the methods to perform the registration of
+functional data and related routines, in basis form as well in discretized form.
+
+"""
+from enum import Enum
+
+import numpy
+import scipy.integrate
+
+
+class Extrapolation(Enum):
+    r"""Enum with extrapolation types. Defines the extrapolation mode for
+        elements outside the domain range.
+    """
+    extrapolation = "extrapolation" #: The values are extrapolated by evaluate.
+    periodic = "periodic" #: Extends the domain range periodically.
+    const = "const" #: Uses the boundary value.
+    slice = "slice" #: Avoids extrapolation restricting the domain.
+
+def mse_decomposition(original_fdata, registered_fdata, warping_function=None,
```

Comment:
Please, change signature to
```
def mse_decomposition(original_fdata, registered_fdata, *, warping_function=None, tfine=None):
```

Reply:
Changed in 455e78a.

Reply:
Is there any reason for `warping_function` to be positional?

Reply:
Because in general the registration will involve warpings functions, and only in particular cases as the shift registration the third argument won't be passed.

Reply:
Ok, it makes sense